alibi_detect/cd/tests/test_cvm_online.py

```diff
@@ -12,29 +12,22 @@
 batch_size = [None, int(n_bootstraps/4)]
 n_features = [1, 3]
 
-tests_cvmdriftonline = list(product(window_sizes, batch_size, n_features))
-n_tests = len(tests_cvmdriftonline)
 
+@pytest.mark.parametrize('window_sizes', window_sizes)
+@pytest.mark.parametrize('batch_size', batch_size)
+@pytest.mark.parametrize('n_feat', n_features)
+def test_cvmdriftonline(window_sizes, batch_size, n_feat, seed):
+    with fixed_seed(seed):
```

Comment:
Noting that the previous version of tests didn't have a fixed seed, presumably it wasn't needed in this setting as test suite has been passing. Is there a need to introduce a fixed seed here as it seems detrimental to the testing for this particular set of tests?

Reply:
P.S. same comment applies to tests below.

Reply:
Mmn yeh I figured it would be good to add since there is randomness in the initialization of these detectors (in `_configure_thresholds`, and in the generation of `x_ref`/`x_h0`/`x_h1`). Although the tests do currently pass without fixing the seed, this doesn't actually mean they pass for any random seed. I seem to recall that when I looked into this before, `np.random.seed`'s set in one test leaked into others. Presumably, this means we have been implicitly fixing the seed in these tests anyway. 

Ideally (IMO), we'd get to a point where any random operations in tests are done inside `with fixed_seed(seed)`'s, then if a new bug is introduced, we can go back and reproduce it with the same random seed.

Reply:
I agree that in case a bug happens then it's valuable to be able to reproduce with the same seed. But, on the other hand, "any random operations in tests done inside `with fixed_seed(seed)`" sounds like the opposite to what we want to do (unless for tests where we compare outputs of the same seed) - as stuff should pass most tests with any seed?

Reply:
Yeh fair point, tests should generally pass with any seed, especially if they are unit type tests. The problem at the moment is we have lots of functional tests where we are testing a detector's predictions and checking things like Expected Runtime (ERT) for online detectors. We probably want more granular unit tests in lots of places...